src/database/queries/sessions.js

```diff
@@ -0,0 +1,295 @@
+		})
+
+		return rowsAffected
+	} catch (error) {
+		return error
+	}
+}
+
+exports.findAll = async (filter, options = {}) => {
+	try {
+		return await Session.findAll({
+			where: filter,
+			...options,
+			raw: true,
+		})
+	} catch (error) {
+		return error
+	}
+}
+
+exports.updateEnrollmentCount = async (sessionId, increment = true) => {
+	try {
+		const options = increment ? { by: 1 } : { by: -1 }
+		const result = this.incrementOrDecrement(
+			{
+				where: { id: sessionId },
+				...options,
+			},
+			'seats_remaining'
+		)
+		return result
+	} catch (error) {
+		return error
+	}
+}
+
+exports.incrementOrDecrement = async (filterWithOptions, incrementFields = []) => {
+	try {
+		return await Session.increment(incrementFields, filterWithOptions)
+	} catch (error) {
+		return error
+	}
+}
+
+exports.getSessionByUserIdAndTime = async (userId, startDate, endDate, sessionId) => {
+	try {
+		let startDateResponse, endDateResponse
+		const query = {
+			mentor_id: userId,
+			status: { [Op.ne]: common.COMPLETED_STATUS },
+		}
+
+		if (startDate) {
+			query.start_date = {
+				[Op.lte]: startDate,
+			}
+			query.end_date = {
+				[Op.gte]: startDate,
+			}
+
+			if (sessionId) {
+				// check if sessionId is truthy (i.e. not undefined or empty)
+				query.id = { [Op.ne]: sessionId }
+			}
+
+			startDateResponse = await this.findAll(query)
+		}
+		if (endDate) {
+			query.start_date = {
+				[Op.lte]: endDate,
+			}
+			query.end_date = {
+				[Op.gte]: endDate,
+			}
+
+			if (sessionId) {
+				// check if sessionId is truthy (i.e. not undefined or empty)
+				query.id = { [Op.ne]: sessionId }
+			}
+
+			endDateResponse = await this.findAll(query)
+		}
+
+		return {
+			startDateResponse: startDateResponse,
+			endDateResponse: endDateResponse,
+		}
+	} catch (error) {
+		return error
+	}
+}
+
+exports.deleteSession = async (filter) => {
+	try {
+		return await Session.destroy({
+			where: filter,
+		})
+	} catch (error) {
+		return error
+	}
+}
+
+exports.updateSession = async (filter, update, options = {}) => {
+	try {
+		return await await Session.update(update, {
+			where: filter,
+			...options,
+		})
+	} catch (error) {
+		console.log(error)
+		return error
+	}
+}
+exports.removeAndReturnMentorSessions = async (userId) => {
+	try {
+		const currentEpochTime = moment().unix()
+		const currentDate = moment()
+		const currentDateTime = moment().format('YYYY-MM-DD HH:mm:ssZ')
+
+		/* const foundSessionOwnerships = await SessionOwnerships.findAll({
+			attributes: ['session_id'],
+			where: {
+				mentor_id: userId,
+			},
+			include: [
+				{
+					model: Session,
+					where: {
+						deleted: false,
+						[Op.or]: [{ startDate: { [Op.gt]: currentEpochTime } }, { status: common.PUBLISHED_STATUS }],
+					},
+					attributes: ['id', 'title'],
+				},
+			],
+		}) */
+		const foundSessionOwnerships = await SessionOwnership.findAll({
```

Comment:
[DISCUSS] If model functions of another table be invovked directly in this file's function?